src/components/Database/DataViewer/NodeActions.tsx

```diff
@@ -110,6 +111,8 @@ export const NodeActions = React.memo<Props>(function NodeActions$({
     onDisplayTypeChange(getNextDisplayType(displayType));
   };
 
+  const potentialQuery = applyQuery(realtimeRef, queryParams || {});
```

Comment:
This query will be a new object each time I think, can you `useMemo` on it?

Reply:
Done.